public/app/algorithms/hash_users.py

```diff
@@ -4,7 +4,11 @@
 # expects sys.argv[1] to be path for json file, formatted as per graasp research
 with open(sys.argv[1]) as json_file:
     allData = json.load(json_file)
-    with open(sys.argv[2], "w+") as anonymizedFile:
+    with open(sys.argv[2], 'w+') as anonymizedFile:
         for action in allData['data']['actions']:
-            action['user'] = hash(action['user'])
-        json.dump(allData, anonymizedFile)
+            if 'user' in action:
+                action['user'] = format(hash(action['user']), 'x')
```

Comment:
by the way, check out https://stackoverflow.com/questions/27522626/hash-function-in-python-3-3-returns-different-results-between-sessions. It turns out that as of Python 3.3 the built-in `hash` function returns a different value for the same input whenever there is a new session. for various reasons this might be problematic in our app. not sure if we are able to load modules yet, but the recommended solution is to use the hashlib module